src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1618,11 +1606,25 @@ void IndexWriter::mergeTerms(bool hasProx) {
 
         match[matchSize++] = queue->pop();
         Term *smallestTerm = match[0]->term;
-        // std::wstring ws = smallestTerm->text();
-        // std::string name = std::string(ws.begin(), ws.end());
-        // std::cout << name << std::endl;
-
         SegmentMergeInfo *top = queue->top();
+        if (infoStream != nullptr) {
+            std::wstring ws = smallestTerm->text();
+            std::wstring f = smallestTerm->field();
+            std::string name = std::string(ws.begin(), ws.end());
+            std::string field = std::string(f.begin(), f.end());
+            message("smallestTerm name: " + name);
+            message("smallestTerm field: " + field);
+
+            if (top != nullptr) {
+                Term* topTerm = top->term;
```

Comment:
Are these wstring and string variables just for debug message? If true, add debug check for performance.

Reply:
infoStream is null by default.
It's a common use in CLucene for debug. There is not a performance issue.